### PR TITLE
Vertx micrometer metrics ubi8.4

### PR DESCRIPTION
**Pull request by vikasgupta8**
created Script to build and validate the package vertx-micrometer-metrics.

### CODE REVIEW

**Comment by vikasgupta8** [comment]
Reopened after oscc approval.

File: v/vertx-micrometer-metrics/vertx-micrometer-metrics_ubi8.4.sh [hunk 1]
```diff
@@ -0,0 +1,107 @@
+# ----------------------------------------------------------------------------
+#
+# Package       : vertx-micrometer-metrics
+# Version       : 3.9.7
+# Source repo   : https://github.com/vert-x3/vertx-micrometer-metrics
+# Tested on     : RHEL UBI 8.4
```
Comment (amitsadaphule) [inline v/vertx-micrometer-metrics/vertx-micrometer-metrics_ubi8.4.sh:6]: Does this mean that the script has been validated on RHEL 8.4 as well as UBI 8.4? The scripts which work on UBI 8.4 almost certainly work on RHEL 8.4, but the reverse is not always true.

File: v/vertx-micrometer-metrics/vertx-micrometer-metrics_ubi8.4.sh [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+# ------- Install dependencies ------
+
+yum -y install git wget make maven gcc-c++ openssl-devel automake autoconf libtool
+yum -y install java-1.8.0-openjdk-devel
+
+#yum -y install gcc-c++.ppc64le
+#yum -y install wget
+#yum -y install git
+#yum install -y openssl-devel.ppc64le
+#yum install -y cmake.ppc64le cmake3.ppc64le
```
Comment (amitsadaphule) [inline v/vertx-micrometer-metrics/vertx-micrometer-metrics_ubi8.4.sh:35]: Please remove these unnecessary/commented installations.

File: v/vertx-micrometer-metrics/vertx-micrometer-metrics_ubi8.4.sh [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+
+dnf install make maven git sudo wget gcc-c++ apr-devel perl openssl-devel automake autoconf libtool -y
```
Comment (amitsadaphule) [inline v/vertx-micrometer-metrics/vertx-micrometer-metrics_ubi8.4.sh:37]: Why are we re-installing these packages which have already been installed on line 28?

File: v/vertx-micrometer-metrics/vertx-micrometer-metrics_ubi8.4.sh [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+
+cd $HOME
+echo `pwd`
+
+# --------- Installing cmake version 3.21.2 -----------------
+echo "--------- Installing cmake version 3.21.2 -----------------"
+
+wget https://github.com/Kitware/CMake/releases/download/v3.21.2/cmake-3.21.2.tar.gz
```
Comment (amitsadaphule) [inline v/vertx-micrometer-metrics/vertx-micrometer-metrics_ubi8.4.sh:45]: Is cmake `3.21.2` a specific requirement here? Does `3.20.2` which is readily available via CentOS AppStream repo not useful here?

File: v/vertx-micrometer-metrics/vertx-micrometer-metrics_ubi8.4.sh [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+#
+# ----------------------------------------------------------------------------
+#!/bin/bash
+
+PACKAGE_NAME=vertx-micrometer-metrics
+PACKAGE_VERSION=3.9.7
+PACKAGE_URL=https://github.com/vert-x3/vertx-micrometer-metrics
+
+mkdir -p /home/tester
+export HOME=/home/tester
```
Comment (amitsadaphule) [inline v/vertx-micrometer-metrics/vertx-micrometer-metrics_ubi8.4.sh:24]: It's not a good idea to override the `HOME` env. Please use a different name.

File: v/vertx-micrometer-metrics/vertx-micrometer-metrics_ubi8.4.sh [hunk 6]
```diff
@@ -0,0 +1,107 @@
...
+export HOME_DIR=/home/tester/$PACKAGE_NAME
+cd $HOME_DIR
+git checkout $PACKAGE_VERSION
+#cd $PACKAGE_NAME
+
+sed -i 's/linux-x86_64/linux-ppcle_64/g' /home/tester/vertx-micrometer-metrics/pom.xml
+
+mvn clean install
+
+echo "-------- Building $PACKAGE_NAME is successful----------------"
```
Comment (amitsadaphule) [inline v/vertx-micrometer-metrics/vertx-micrometer-metrics_ubi8.4.sh:106]: We're not using `set -e` in the script. How do we know that the previous command succeeded?

**Review by amitsadaphule — CHANGES_REQUESTED** [review]

File: v/vertx-micrometer-metrics/vertx-micrometer-metrics_ubi8.4.sh (hunk 1 above)
Comment (vikasgupta8) [reply v/vertx-micrometer-metrics/vertx-micrometer-metrics_ubi8.4.sh]: It was tested on UBI 8.4

File: v/vertx-micrometer-metrics/vertx-micrometer-metrics_ubi8.4.sh (hunk 2 above)
Comment (vikasgupta8) [reply v/vertx-micrometer-metrics/vertx-micrometer-metrics_ubi8.4.sh]: removed comments.

File: v/vertx-micrometer-metrics/vertx-micrometer-metrics_ubi8.4.sh (hunk 4 above)
Comment (vikasgupta8) [reply v/vertx-micrometer-metrics/vertx-micrometer-metrics_ubi8.4.sh]: Removed reinstallation of cmake, tested with cmake-3.20.2 and it worked.

File: v/vertx-micrometer-metrics/vertx-micrometer-metrics_ubi8.4.sh (hunk 3 above)
Comment (vikasgupta8) [reply v/vertx-micrometer-metrics/vertx-micrometer-metrics_ubi8.4.sh]: removed packages reinstallation.

File: v/vertx-micrometer-metrics/vertx-micrometer-metrics_ubi8.4.sh (hunk 5 above)
Comment (vikasgupta8) [reply v/vertx-micrometer-metrics/vertx-micrometer-metrics_ubi8.4.sh]: renamed to HOME_DIR

File: v/vertx-micrometer-metrics/vertx-micrometer-metrics_ubi8.4.sh (hunk 6 above)
Comment (vikasgupta8) [reply v/vertx-micrometer-metrics/vertx-micrometer-metrics_ubi8.4.sh:87]: It fails while building netty-tcnative(only BoringSSL). But that does not impact the build/test of vertx-micrometer-metrics.
set -e will stop script execution.

Comment (amitsadaphule) [reply v/vertx-micrometer-metrics/vertx-micrometer-metrics_ubi8.4.sh:87]: Does that mean that the test execution, etc completes successfully?

Comment (vikasgupta8) [reply v/vertx-micrometer-metrics/vertx-micrometer-metrics_ubi8.4.sh:87]: Yes the test result is successful.

**Review by amitsadaphule — APPROVED** [review]
LGTM

**Pull request merged**